Single sample classification

komalsrathi opened the pull request:
I have made a few changes in the code to allow the capability for single-sample subtyping - which was long due. This will enable the dev team to be able to run the subtyping on each new sample individually without having to run it on existing classified samples.

Here are some tests that I have run to make sure that the single-sample subtyping output matches the multi-sample subtyping output:


**Dataset 1: GSE109401**

```
library(medulloPackage)
data("exprs_109401")

# classify multiple samples together
pred_109401 <- classify(exprs = exprs_109401)

# classify one sample at a time
total <- data.frame()
for(i in 1:ncol(exprs_109401)){
  exprs <- exprs_109401[,i,drop = F]
  pred <- classify(exprs = exprs)
  total <- rbind(total, pred)
}

# check if the classification output df is identical?
identical(pred_109401, total)
[1] TRUE
```

**Dataset 2: OpenPBTA**

```
setwd('~/Projects/OpenPBTA-analysis/analyses/molecular-subtyping-MB/')
suppressPackageStartupMessages({
  library(tidyverse)
  library(medulloPackage)
})

# set directories
root_dir <- rprojroot::find_root(rprojroot::has_dir(".git"))
data_dir <- file.path(root_dir, "data")
clin <- read_tsv(file.path(root_dir, "data", "pbta-histologies-base.tsv"))
path_dx_list <- file.path("input", "mb_subtyping_path_dx_strings.json") %>%
  jsonlite::fromJSON()
clin.mb  <- clin %>%
  # Inclusion on the basis of strings in pathology_diagnosis and
  # if 'Other', then based on pathology_free_text_diagnosis
  filter(pathology_diagnosis %in% path_dx_list$exact_path_dx |
           (pathology_diagnosis == "Other" & str_detect(str_to_lower(pathology_free_text_diagnosis), 
                                                        paste0(path_dx_list$include_free_text, collapse = "|"))))

# MB RNA-seq samples only
clin.mb.rnaseq <- clin.mb %>%
  dplyr::filter(experimental_strategy == "RNA-Seq")

# fpkm dataset subsetted to MB RNA-seq samples
polya.mb <- file.path(root_dir, "data", "pbta-gene-expression-rsem-fpkm-collapsed.polya.rds") %>%
  readRDS() %>%
  dplyr::select(any_of(clin.mb.rnaseq$Kids_First_Biospecimen_ID))
stranded.mb <- file.path(root_dir, "data","pbta-gene-expression-rsem-fpkm-collapsed.stranded.rds") %>%
  readRDS() %>%
  dplyr::select(any_of(clin.mb.rnaseq$Kids_First_Biospecimen_ID))

# combine both polyA and stranded samples
all.exprs.mb <- polya.mb %>%
  rownames_to_column("gene") %>%
  inner_join(stranded.mb %>% rownames_to_column("gene"), by = "gene") %>%
  column_to_rownames('gene') %>%
  as.matrix()

# log2 transformation
all.exprs.mb <- log2(all.exprs.mb + 1)
dim(all.exprs.mb) 
# [1] 46303   122

# complete data
res <- medulloPackage::classify(exprs = all.exprs.mb)

# per sample classification
total <- data.frame()
for(i in 1:ncol(all.exprs.mb)){
  exprs <- all.exprs.mb[,i,drop = F]
  pred <- medulloPackage::classify(exprs = exprs)
  total <- rbind(total, pred)
}

# are both classifications identical?
> identical(res, total)
[1] TRUE

# output of a few samples
# multi-sample classification
> head(res)
                 sample best.fit      p.value
BS_S6Q7NKA3 BS_S6Q7NKA3      SHH 9.571056e-03
BS_09Z7TC35 BS_09Z7TC35   Group3 3.616578e-11
BS_1AYRM596 BS_1AYRM596   Group4 9.166351e-03
BS_1BWP5MCT BS_1BWP5MCT   Group3 7.040973e-10
BS_1QXEC43H BS_1QXEC43H      SHH 2.053227e-04
BS_1TWCV047 BS_1TWCV047   Group3 1.832252e-05

# single-sample classification
> head(total)
                 sample best.fit      p.value
BS_S6Q7NKA3 BS_S6Q7NKA3      SHH 9.571056e-03
BS_09Z7TC35 BS_09Z7TC35   Group3 3.616578e-11
BS_1AYRM596 BS_1AYRM596   Group4 9.166351e-03
BS_1BWP5MCT BS_1BWP5MCT   Group3 7.040973e-10
BS_1QXEC43H BS_1QXEC43H      SHH 2.053227e-04
BS_1TWCV047 BS_1TWCV047   Group3 1.832252e-05


```

komalsrathi committed:
single sample classification

komalsrathi commented:
I have tested this on 2 datasets as mentioned above. It works as expected - if you do single sample classification vs if the sample is part of the larger cohort, the classification does not change. 

Please review if you can and I can merge this for use by the devops team.

jharenza commented:
@komalsrathi before merging this, can you create a release for the current repo, and then create a new release once merged, that way it can be tracked and installs tied neatly to releases. thanks!

komalsrathi committed:
update version

komalsrathi commented:
I have added a release to the existing master branch. Also updated the `DESCRIPTION` file with a new updated version for the next release once this is merged. Thanks.

komalsrathi commented:
@sickler-alex would you be able to review this? Thank you!

sickler-alex commented on R/calcScore.R:
```diff
@@ -53,7 +53,7 @@ calcScore <- function(myMat = NULL, mySetsUp = NULL) {
 
   getScoreSet <- function(x, myMat = myMat) {
-    return(colMeans(myMat[rownames(myMat) %in% x,]))
+    return(colMeans(myMat[rownames(myMat) %in% x,,drop = F]))
```
What does the double comma do here?
